modules/gapi/test/infer/gapi_infer_onnx_test.cpp

```diff
@@ -264,15 +381,82 @@ TEST(ONNX, Infer2ROIList)
     comp.apply(cv::gin(in_mat1, rois),
                cv::gout(out_gapi),
                cv::compile_args(cv::gapi::networks(net)));
+    // Validate
+    validate();
+}
 
+TEST_F(ONNXtest, InferDynamicInputTensor)
+{
+    useModel("object_detection_segmentation/tiny-yolov2/model/tinyyolov2-8");
+    // Create tensor
+    const cv::Mat rand_mat = initMatrixRandU(CV_32FC3, cv::Size{416, 416});
+    const std::vector<int> dims = {1, rand_mat.channels(), rand_mat.rows, rand_mat.cols};
+    cv::Mat tensor(dims, CV_32F, rand_mat.data);
+    const cv::Mat in_tensor = tensor / 255.f;
+    // ONNX_API code
+    infer<float>(in_tensor, out_onnx.front());
+    // G_API code
+    G_API_NET(YoloNet, <cv::GMat(cv::GMat)>, "YoloNet");
+    cv::GMat in;
+    cv::GMat out = cv::gapi::infer<YoloNet>(in);
+    cv::GComputation comp(cv::GIn(in), cv::GOut(out));
+    auto net = cv::gapi::onnx::Params<YoloNet>{model_path}
+        .cfgPostProc({cv::GMatDesc{CV_32F, {1,125,13,13}}}, remap_yolo)
+        .cfgOutputLayers({"out"});
+    comp.apply(cv::gin(in_tensor),
+               cv::gout(out_gapi.front()),
+               cv::compile_args(cv::gapi::networks(net)));
     // Validate
-    for (size_t i = 0; i < rois.size(); ++i) {
-        ASSERT_EQ(1000u, out_onnx[i].total());
-        ASSERT_EQ(1000u, out_gapi[i].total());
-        normAssert(out_onnx[i], out_gapi[i], "Test classification output");
-    }
+    validate();
+}
+
+TEST_F(ONNXGRayScaleTest, InferImage)
+{
+    useModel("body_analysis/emotion_ferplus/model/emotion-ferplus-8");
+    // ONNX_API code
+    cv::Mat prep_mat;
+    preprocess(in_mat1, prep_mat);
+    infer<float>(prep_mat, out_onnx.front());
+    // G_API code
+    G_API_NET(EmotionNet, <cv::GMat(cv::GMat)>, "emotion-ferplus");
+    cv::GMat in;
+    cv::GMat out = cv::gapi::infer<EmotionNet>(in);
+    cv::GComputation comp(cv::GIn(in), cv::GOut(out));
+    auto net = cv::gapi::onnx::Params<EmotionNet> { model_path }
+        .cfgNormalize({false}); // model accepts 0..255 range in FP32;
+    comp.apply(cv::gin(in_mat1),
+               cv::gout(out_gapi.front()),
+               cv::compile_args(cv::gapi::networks(net)));
+    // Validate
+    validate();
 }
 
+TEST_F(ONNXtest, InferMultOutput)
+{
+    useModel("object_detection_segmentation/ssd-mobilenetv1/model/ssd_mobilenet_v1_10");
+    // ONNX_API code
+    const auto prep_mat = in_mat1.reshape(1, {1, in_mat1.rows, in_mat1.cols, in_mat1.channels()});
+    infer<uint8_t>({prep_mat}, out_onnx);
+    // G_API code
+    using OUT = std::tuple<cv::GMat, cv::GMat, cv::GMat, cv::GMat>;
```

Comment:
Let it be `GMobileNetOut`, for example. Or `GSSDOut`

Reply:
Fixed.